python_modules/libraries/dagster-dg/dagster_dg/docs.py

```diff
@@ -184,6 +205,87 @@ def open_html_in_browser(html_content: str) -> None:
     webbrowser.open(f"file://{temp_file_path}")
 
 
+# code blocks are of the format ..code-block:: <language>
+import re
+
+# match lines of the form
+# ..code-block:: <language>
+#   foo
+#   bar
+#   baz
+CODE_BLOCK_REGEX = re.compile(r"(\.\. code-block::.*?\n(:?[\s\S]+))", re.DOTALL | re.MULTILINE)
+
+
+def process_description(description: str) -> str:
+    # replace code blocks with <code> tags
+    description = CODE_BLOCK_REGEX.sub(r"", description)
+    return description.replace("\n", "<br>")
+
+
+def markdown_for_json_schema(
+    key: str,
+    json_schema: Mapping[str, Any],
+    subschema: Mapping[str, Any],
+    anyof_parent_subschema: Optional[Mapping[str, Any]] = None,
+    indent: int = 0,
+) -> str:
+    """Produces a nested markdown list of the subschema, including component-author-provided description and examples.
+    Uses <details> blocks to collapse nested fields by default.
+    """
+    subschema = _dereference_schema(json_schema, subschema)
+
+    if "anyOf" in subschema:
+        # TODO: handle anyOf fields more gracefully, for now just choose first option
+        return markdown_for_json_schema(
+            key, json_schema, subschema["anyOf"][0], anyof_parent_subschema=subschema, indent=indent
+        )
+
+    objtype = subschema["type"]
+
+    children = ""
+    if objtype == "object":
+        children = "\n".join(
+            [
+                markdown_for_json_schema(k, json_schema, v, indent=indent + 2)
+                for k, v in subschema.get("properties", {}).items()
+            ]
+        )
+    elif objtype == "array":
+        return markdown_for_json_schema(key, json_schema, subschema["items"])
+
+    description = process_description(
+        subschema.get("description", "") or (anyof_parent_subschema or {}).get("description", "")
+    )
+    children_segment = (
+        f"""<details>
+<summary>Subfields</summary>
+\n\n
+<ul>
+{children}
+</ul>
+</details>
+    """
```

Comment:
pls `textwrap.dedent(...).strip()`. Then you can indent with rest of python code